cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/BasicPushStrategyTest.java

```diff
@@ -35,115 +35,144 @@
 @Category(RequireRunningCouchDB.class)
 public class BasicPushStrategyTest extends ReplicationTestBase {
 
-    PushConfiguration currentSetting = null;
-    BasicPushStrategy replicator;
+    // some helpers...
 
-    @Before
-    public void setUp() throws Exception {
-        super.setUp();
-        currentSetting = createPushReplicationSetting();
+    // we use this utility method rather than ReplicationTestBase.push() because some
+    // methods want to make assertions on the BasicPushStrategy after running the replication
+    private void push(BasicPushStrategy replicator, int expectedDocs) throws Exception {
+        TestStrategyListener listener = new TestStrategyListener();
+        replicator.eventBus.register(listener);
+        replicator.run();
+        Assert.assertTrue(listener.finishCalled);
+        Assert.assertFalse(listener.errorCalled);
+        Assert.assertEquals(expectedDocs, listener.documentsReplicated);
     }
 
-    private PushConfiguration createPushReplicationSetting() {
-        return new PushConfiguration(PushConfiguration.DEFAULT_CHANGES_LIMIT_PER_BATCH,
-                PushConfiguration.DEFAULT_MAX_BATCH_COUNTER_PER_RUN, 1, PushConfiguration.DEFAULT_PUSH_ATTACHMENTS_INLINE);
+    private void assertPushReplicationStatus(BasicPushStrategy replicator, int documentCounter, int batchCounter, String lastSequence) throws Exception {
+        Assert.assertEquals("DocumentRevisionTree counter", documentCounter, replicator
+                .getDocumentCounter());
```

Comment:
Again, I would consider splitting the line breaks on the full argument.  Same as https://github.com/cloudant/sync-android/pull/222/files#r47793595